### ISSUE TITLE
Consider removing the .onLoad functionality 

### ISSUE THREAD

**Issue by dshemetov** [gaborcsardi/dotenv#14]
The function in #13 is especially useful when you want to load variables, but not overwrite your env vars. This is moot due to the `.onLoad` function in this package, which eagerly loads the env vars. FWIW, the [python-dotenv package](https://saurabh-kumar.com/python-dotenv/#getting-started) also separates package import and the env var load.

I understand if the migration/backwards incompatibility is not worth it for the maintainers, but just want to open this idea up for your consideration at some point in the future.

**Comment by gaborcsardi** [comment]
Yeah, that would be great, but it would also break practically every project that uses this package, so we cannot do it at this point, at least not by default.

You could opt in into the new behavior by setting an env var. Not as nice, but possible ok?
